extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.setup.js

```diff
@@ -54,15 +57,37 @@ angular.module('PaperUI.controllers.setup', []).controller('SetupPageController'
                 'enableChannels' : !$scope.advancedMode
             }, result.label).$promise.then(function() {
                 thingRepository.setDirty(true);
-                toastService.showDefaultToast('Thing added.', 'Show Thing', 'configuration/things/view/' + thingUID);
                 var thingType = thingTypeRepository.find(function(thingType) {
                     return thingTypeUID === thingType.UID;
                 });
 
-                if (thingType && thingType.bridge) {
-                    $scope.navigateTo('setup/search/' + thingUID.split(':')[0]);
+                var configRequired = false;
+                var thing = thingRepository.find(function(thing) {
+                    return thing.UID === thingUID;
+                });
+                if (thing && thing.statusInfo.status === 'OFFLINE' && thing.statusInfo.statusDetail === 'CONFIGURATION_ERROR') {
+                    configRequired = true;
+                } else if (thing && thing.statusInfo.status !== 'ONLINE') {
+                    for ( var key in thing.configuration) {
+                        if (thing.configuration.hasOwnProperty(key)) {
+                            if (thing.configuration[key] === '') {
```

Comment:
Don't you have the information regarding the required properties in your `thing.configuration` object? I mean the paperUI shows these fields in red, so the information must be there somehow :)

As @htreu suggested in https://github.com/eclipse/smarthome/issues/3657#issuecomment-349871679 you could then only redirect the user if a required field is missing instead of always redirecting the user if a field is empty.

Reply:
Hello and thanks for your answer. Sure, it was also my intention. Sadly this information is not available in JS. Configuration Object in this layer is a simple Map. In my case:
`{host: "127.0.0.1", protocol: "http", password: "", poll: 5, port: "5000", username: ""}`
here password is mandatory and username not (for this test only).

Reply:
There is a service to retrieve config descriptions for a given config description URI. Check out the `configDescriptionService` and review its usages for an example.
With the the result from this service you get the meta data to check for required fields.

Reply:
Thanks for this tip!! 
I've removed my old approach almost entirely and replaced it with looking into `configDescriptionService` of this `thing-type` and checking for being empty and `required`.  
If the call to `configDescriptionService` fails, it falls back to old behavior.